### PR TITLE
update for raspbian buster

### PR DESCRIPTION
**Pull request by MarcTreySonos**
fixed the dependancies list and installation 

there is an issue when deploying the addon manually as the code still fetches from github

```2019-07-26 17:21:21.919 INFO   : Fetching add-on https://github.com/andrenatal/voice-addon/releases/download/1.0.0/voice-addon-1.0.0.tgz as /tmp/JuZmGc/voice-addon.tar.gz```

### CODE REVIEW

**Comment by kgiori** [comment]
@mtrey -- Thanks for the quick offer of an installation dependency workaround for the 0.9.* release!

I had a chance to try this PR today. For an RPi3 gateway that was updated OTA from 0.8.1 to 0.9.1, meaning that the RPi build was originally on Debian/Raspbian Stretch, the install script failed to run with the error listed below.

Next I'll try the voice-addon installation on a "fresh" 0.9.1 install (Debian/Raspbian Buster). Is it possible to have this PR address both updated gateways and new gateways (Stretch and Buster)? 

Also, what would be a better long-term approach for Snips package integration to avoid these types of dependency errors? 

```2019-07-29 20:53:08.311 INFO   : Loading add-on: voice-addon
2019-07-29 20:53:09.124 INFO   : voice-addon: Opening database: /home/pi/.mozilla-iot/config/db.sqlite3
2019-07-29 20:53:09.171 INFO   : voice-addon: Loading add-on for voice-addon from /home/pi/.mozilla-iot/addons/voice-addon
2019-07-29 20:53:09.744 INFO   : voice-addon: microphone USB
2019-07-29 20:53:09.747 INFO   : voice-addon: speaker USB
2019-07-29 20:53:09.748 INFO   : voice-addon: writing asound.conf
2019-07-29 20:53:09.766 INFO   : voice-addon: asound.conf written
2019-07-29 20:53:09.796 INFO   : voice-addon: addedEvent wakeword { description: 'A wakeword was deteced', type: 'string' }
2019-07-29 20:53:09.799 INFO   : voice-addon: addedEvent speechinput { description: 'A voice command was detected', type: 'string' }
2019-07-29 20:53:09.803 INFO   : voice-addon: addedEvent command { description: 'A web thing command was executed',
2019-07-29 20:53:09.806 INFO   : voice-addon:   type: 'string' }
2019-07-29 20:53:09.811 INFO   : voice-addon: addedEvent training { description: 'Wakeword training started', type: 'string' }
2019-07-29 20:53:10.043 INFO   : voice-addon: Error executing install_script.sh E: Command line option --allow-releaseinfo-change is not understood in combination with the other options```

**Comment by kgiori** [comment]
@mtrey -- below is a log snippet from what I see when I enable the PR-updated addon to a fresh 0.9.1 image install.

```2019-07-29 22:39:17.820 INFO   : voice-addon: Opening database: /home/pi/.mozilla-iot/config/db.sqlite3
2019-07-29 22:39:17.863 INFO   : voice-addon: Loading add-on for voice-addon from /home/pi/.mozilla-iot/addons/voice-addon
2019-07-29 22:39:18.376 INFO   : voice-addon: microphone USB
2019-07-29 22:39:18.378 INFO   : voice-addon: speaker USB
2019-07-29 22:39:18.380 INFO   : voice-addon: writing asound.conf
2019-07-29 22:39:18.395 INFO   : voice-addon: asound.conf written
2019-07-29 22:39:18.425 INFO   : voice-addon: addedEvent wakeword { description: 'A wakeword was deteced', type: 'string' }
2019-07-29 22:39:18.428 INFO   : voice-addon: addedEvent speechinput { description: 'A voice command was detected', type: 'string' }
2019-07-29 22:39:18.430 INFO   : voice-addon: addedEvent command { description: 'A web thing command was executed',
2019-07-29 22:39:18.432 INFO   : voice-addon:   type: 'string' }
2019-07-29 22:39:18.437 INFO   : voice-addon: addedEvent training { description: 'Wakeword training started', type: 'string' }
2019-07-29 22:39:19.181 INFO   : voice-addon: DATA snips_installation: Get:1 http://archive.raspberrypi.org/debian buster InRelease [25.1 kB]
2019-07-29 22:39:19.183 INFO   : voice-addon: 
2019-07-29 22:39:19.302 INFO   : voice-addon: DATA snips_installation: Get:2 http://raspbian.raspberrypi.org/raspbian buster InRelease [15.0 kB]
2019-07-29 22:39:19.304 INFO   : voice-addon: 
2019-07-29 22:39:20.024 INFO   : voice-addon: DATA snips_installation: Get:3 http://raspbian.raspberrypi.org/raspbian buster/main armhf Packages [13.0 MB]
2019-07-29 22:39:20.027 INFO   : voice-addon: 
2019-07-29 22:39:20.330 INFO   : voice-addon: DATA snips_installation: Get:4 http://archive.raspberrypi.org/debian buster/main armhf Packages [212 kB]
2019-07-29 22:39:20.333 INFO   : voice-addon: 
2019-07-29 22:39:28.757 INFO   : voice-addon: Error calling things api: SyntaxError: Unexpected end of JSON input
2019-07-29 22:39:38.697 INFO   : voice-addon: Error calling things api: SyntaxError: Unexpected end of JSON input
2019-07-29 22:39:48.697 INFO   : voice-addon: Error calling things api: SyntaxError: Unexpected end of JSON input
2019-07-29 22:39:58.700 INFO   : voice-addon: Error calling things api: SyntaxError: Unexpected end of JSON input
2019-07-29 22:40:08.705 INFO   : voice-addon: Error calling things api: SyntaxError: Unexpected end of JSON input
2019-07-29 22:40:09.567 INFO   : voice-addon: DATA snips_installation: Get:5 http://raspbian.raspberrypi.org/raspbian buster/contrib armhf Packages [58.7 kB]
2019-07-29 22:40:09.570 INFO   : voice-addon: 
2019-07-29 22:40:18.708 INFO   : voice-addon: Error calling things api: SyntaxError: Unexpected end of JSON input
2019-07-29 22:40:19.196 INFO   : voice-addon: DATA snips_installation: Fetched 13.3 MB in 60s (220 kB/s)
2019-07-29 22:40:19.199 INFO   : voice-addon: Reading package lists...
2019-07-29 22:40:28.716 INFO   : voice-addon: Error calling things api: SyntaxError: Unexpected end of JSON input
2019-07-29 22:40:28.725 INFO   : voice-addon: DATA snips_installation: 
2019-07-29 22:40:28.730 INFO   : voice-addon: 
2019-07-29 22:40:29.218 INFO   : voice-addon: DATA snips_installation: Reading package lists...
2019-07-29 22:40:29.277 INFO   : voice-addon: DATA snips_installation: 
2019-07-29 22:40:29.281 INFO   : voice-addon: 
2019-07-29 22:40:32.996 INFO   : voice-addon: DATA snips_installation: Building dependency tree...
2019-07-29 22:40:33.750 INFO   : voice-addon: DATA snips_installation: 
2019-07-29 22:40:33.752 INFO   : voice-addon: Reading state information...
2019-07-29 22:40:34.262 INFO   : voice-addon: DATA snips_installation: 
2019-07-29 22:40:34.265 INFO   : voice-addon: 
2019-07-29 22:40:37.729 INFO   : voice-addon: DATA snips_installation: dirmngr is already the newest version (2.2.12-1+rpi1).
2019-07-29 22:40:37.731 INFO   : voice-addon: 0 upgraded, 0 newly installed, 0 to remove and 19 not upgraded.
2019-07-29 22:40:37.733 INFO   : voice-addon: 
2019-07-29 22:40:38.713 INFO   : voice-addon: Error calling things api: SyntaxError: Unexpected end of JSON input
2019-07-29 22:40:39.388 INFO   : voice-addon: Error executing install_script.sh Warning: apt-key output should not be parsed (stdout is not a terminal)
2019-07-29 22:40:39.391 INFO   : voice-addon: 
2019-07-29 22:40:39.456 INFO   : voice-addon: DATA snips_installation: Executing: /tmp/apt-key-gpghome.M3LM0QGw5u/gpg.1.sh --keyserver pgp.mit.edu --recv-keys D4F50CDCA10A2849
2019-07-29 22:40:39.459 INFO   : voice-addon: 
2019-07-29 22:40:40.145 INFO   : voice-addon: Error executing install_script.sh gpg: key D4F50CDCA10A2849: public key "Snips Raspbian distribution <infra@snips.ai>" imported
2019-07-29 22:40:40.147 INFO   : voice-addon: 
2019-07-29 22:40:40.158 INFO   : voice-addon: Error executing install_script.sh gpg: Total number processed: 1
2019-07-29 22:40:40.161 INFO   : voice-addon: gpg:               imported: 1
2019-07-29 22:40:40.163 INFO   : voice-addon: 
2019-07-29 22:40:40.998 INFO   : voice-addon: DATA snips_installation: Hit:1 http://archive.raspberrypi.org/debian buster InRelease
2019-07-29 22:40:41.000 INFO   : voice-addon: 
2019-07-29 22:40:41.003 INFO   : voice-addon: DATA snips_installation: Hit:2 http://raspbian.raspberrypi.org/raspbian buster InRelease
2019-07-29 22:40:41.004 INFO   : voice-addon: 
2019-07-29 22:40:41.120 INFO   : voice-addon: DATA snips_installation: Get:3 https://raspbian.snips.ai/stretch stable InRelease [2,334 B]
2019-07-29 22:40:41.121 INFO   : voice-addon: 
2019-07-29 22:40:42.324 INFO   : voice-addon: DATA snips_installation: Get:4 https://raspbian.snips.ai/stretch stable/main armhf Packages [49.3 kB]
2019-07-29 22:40:42.325 INFO   : voice-addon: 
2019-07-29 22:40:42.545 INFO   : voice-addon: DATA snips_installation: Fetched 51.6 kB in 2s (28.1 kB/s)
2019-07-29 22:40:42.547 INFO   : voice-addon: Reading package lists...
2019-07-29 22:40:46.948 INFO   : voice-addon: DATA snips_installation: 
2019-07-29 22:40:46.950 INFO   : voice-addon: 
2019-07-29 22:40:48.716 INFO   : voice-addon: Error calling things api: SyntaxError: Unexpected end of JSON input
2019-07-29 22:40:56.016 INFO   : voice-addon: Error executing install_script.sh --2019-07-29 22:40:56--  http://ftp.fr.debian.org/debian/pool/non-free/s/svox/libttspico-data_1.0+git20130326-9_armhf.deb
2019-07-29 22:40:56.017 INFO   : voice-addon: 
2019-07-29 22:40:56.059 INFO   : voice-addon: Error executing install_script.sh Resolving ftp.fr.debian.org (ftp.fr.debian.org)... 
2019-07-29 22:40:56.234 INFO   : voice-addon: Error executing install_script.sh 212.27.32.66, 2a01:e0c:1:1598::2
2019-07-29 22:40:56.236 INFO   : voice-addon: 
2019-07-29 22:40:56.238 INFO   : voice-addon: Error executing install_script.sh Connecting to ftp.fr.debian.org (ftp.fr.debian.org)|212.27.32.66|:80... 
2019-07-29 22:40:56.374 INFO   : voice-addon: Error executing install_script.sh connected.
2019-07-29 22:40:56.376 INFO   : voice-addon: 
2019-07-29 22:40:56.378 INFO   : voice-addon: Error executing install_script.sh HTTP request sent, awaiting response... 
2019-07-29 22:40:56.516 INFO   : voice-addon: Error executing install_script.sh 404 Not Found
2019-07-29 22:40:56.519 INFO   : voice-addon: 
2019-07-29 22:40:56.521 INFO   : voice-addon: Error executing install_script.sh 2019-07-29 22:40:56 ERROR 404: Not Found.
2019-07-29 22:40:56.522 INFO   : voice-addon: 
2019-07-29 22:40:56.524 INFO   : voice-addon: 
2019-07-29 22:40:58.720 INFO   : voice-addon: Error calling things api: SyntaxError: Unexpected end of JSON input
2019-07-29 22:41:08.726 INFO   : voice-addon: Error calling things api: SyntaxError: Unexpected end of JSON input```

File: deps/install_deps.sh [hunk 1]
```diff
@@ -44,6 +39,23 @@ install() {
     		exit 1
 	fi
 
+	sudo apt-get update --allow-releaseinfo-change
+	sudo apt-get install -y dirmngr
+	sudo bash -c 'echo "deb https://raspbian.snips.ai/stretch stable main" > /etc/apt/sources.list.d/snips.list'
+	sudo apt-key adv --keyserver pgp.mit.edu --recv-keys D4F50CDCA10A2849
+	sudo apt-get update
+
+	wget http://ftp.fr.debian.org/debian/pool/non-free/s/svox/libttspico-data_1.0+git20130326-9_armhf.deb
```
Comment (mrstegeman) [inline deps/install_deps.sh:48]: This should be `_all.deb`, not `_armhf.deb`.

File: deps/install_deps.sh [hunk 2]
```diff
@@ -44,6 +39,23 @@ install() {
...
+	wget http://ftp.fr.debian.org/debian/pool/non-free/s/svox/libttspico-utils_1.0+git20130326-9_armhf.deb
+	wget http://ftp.fr.debian.org/debian/pool/non-free/s/svox/libttspico0_1.0+git20130326-9_armhf.deb
+	sudo dpkg -i libttspico-data_1.0+git20130326-9_all.deb 
+	sudo dpkg -i libttspico0_1.0+git20130326-9_armhf.deb 
+	sudo dpkg -i libttspico-utils_1.0+git20130326-9_armhf.deb 
+	rm libttspico*.deb
+
+	sudo apt install libgfortran3 || true
+	sudo apt install libatlas3-base=3.10.3-8+rpi1
```
Comment (mrstegeman) [inline deps/install_deps.sh:57]: These need a `-y`.

**Review by mrstegeman — CHANGES_REQUESTED** [review]

**Comment by andrenatal** [comment]
@ mtrey The ideal here is to have a single .deb installed containing all packages and dependencies necessary in order to have the system running. Could you provide one per platform for us, please?

**Comment by MarcTreySonos** [comment]
closing this one in favor of https://github.com/andrenatal/voice-addon/pull/36

**Pull request closed without merging**